packages/shopping/src/controllers/user.controller.ts

```diff
@@ -260,4 +261,46 @@ export class UserController {
 
     return {token};
   }
+
+  @put('/users/password-reset', {
+    responses: {
+      '200': {
+        description: 'The updated user profile',
+        content: {
+          'application/json': {
+            schema: UserProfileSchema,
+          },
+        },
+      },
+    },
+  })
+  async passwordReset(
+    @requestBody(PasswordResetRequestBody) credentials: Credentials,
+  ): Promise<User> {
+    const {email, password} = credentials;
+    // ensure the user exists once found user can change password.
+    // In a more complex scenario a link or OTP will be sent to the
+    // user in order to verify their true identity
+    const user = await this.userRepository.findOne({
+      where: {email: email},
+    });
+
+    if (!user) {
+      throw new HttpErrors.UnprocessableEntity('User account not found');
+    }
+
+    // ensure a valid email value and password value
+    validateCredentials(_.pick(credentials, ['email', 'password']));
+
+    // encrypt the new password
+    const passwordHash = await this.passwordHasher.hashPassword(password);
+
+    // reset the password
+    await this.userRepository
+      .userCredentials(user.id)
+      .patch({password: passwordHash});
+
+    // return the user
+    return this.userRepository.findById(user.id);
```

Comment:
@mrmodise seems unnecessary to do another database operation here since you already have the user model

Reply:
Agreed and I highlighted this in the comments. I can do the full blown solution but I thought it would be an overkill for a POC

Reply:
Updated accordingly